### ISSUE TITLE
Recover from failures. 

### ISSUE THREAD

**Issue by a-nldisr** [mesos/elasticsearch#424]
We have reviewed various Mesos frameworks and one of the questions we always asked ourselves: What happens in case of datacenter failures or an entire cluster failure? (for example a human error or a simple stop and start of the Cluster) 

Usually there is no easy way to get a previous running cluster/framework on Mesos back on the previous nodes to ensure data recovered and till this data is recovered from sandboxes/volumes and imported back in the cluster, the clusters data is inconsistent. 

We now look at deploying clusters through Aurora by using attributes and constraints and by taking a look at using the ElasticSearch HDFS snapshot plugin. Perhaps there are other idea's to solve dealing with datacenter/cluster failures/reboots. 

At today's software circus i had a short conversation with @frankscholten regarding datacenter failures, Frank asked me to open an issue to take scenario's like these into consideration. 

In short:  How can you recover from datacenter/cluster failures/stops with the ElasticSearch/Mesos framework?

**Comment by philwinder** [comment]
Solution: Use external volumes! :-)

**Comment by a-nldisr** [comment]
Does that gaurantee you always land back on the hosts with these volumes? Or do we have to recover manually the data?

**Comment by philwinder** [comment]
@a-nldisr: The volumes are external to the cluster. They are not on the same machines as the tasks. When tasks start, the volumes are automatically mounted/moved to the same location as the new task.

So no, it does not guarantee that you will start the task on the same host. And no, you don't have to manually recover data. That is done for you automatically.
